features/authentication/authentication.api.ts

```diff
@@ -35,23 +57,55 @@ export const createUser = async (args: {
   inboxId: string;
   profile: {
     name: string;
+    username: string;
     avatar?: string;
   };
 }): Promise<CreateUserResponse> => {
   const { privyUserId, smartContractWalletAddress, inboxId, profile } = args;
 
   try {
-    const response = await api.post<CreateUserResponse>("/api/v1/users", {
+    // Build the request payload
+    const requestPayload = {
       privyUserId,
       device: buildDeviceMetadata(),
       identity: {
         privyAddress: smartContractWalletAddress,
         xmtpId: inboxId,
       },
       profile,
-    });
+    };
+
+    // Validate request payload
+    const validationResult = createUserRequestSchema.safeParse(requestPayload);
+    if (!validationResult.success) {
+      logger.error(
+        `[createUser] Request validation failed:
+        ${JSON.stringify(validationResult.error.errors, null, 2)}
+        Payload: ${JSON.stringify(requestPayload, null, 2)}`
+      );
+      throw new Error("Invalid request data");
+    }
+
+    // Make the API call with validated data
+    const response = await api.post<CreateUserResponse>(
+      "/api/v1/users",
+      validationResult.data
+    );
+
+    // Validate response
+    const responseValidation = createUserResponseSchema.safeParse(
+      response.data
+    );
+    if (!responseValidation.success) {
+      logger.error(
+        `[createUser] Response validation failed:
+        ${JSON.stringify(responseValidation.error.errors, null, 2)}
+        Response: ${JSON.stringify(response.data, null, 2)}`
+      );
+      throw new Error("Invalid response data from server");
```

Comment:
something I've been doing is not throwing here and at the end just reutrn the data. Because we might have a slight data validataion error but we don'T want to stop users from using the app 100%